homeassistant/components/maxcube.py

```diff
@@ -39,6 +41,7 @@ def setup(hass, config):
 
     host = config.get(DOMAIN).get(CONF_HOST)
     port = config.get(DOMAIN).get(CONF_PORT)
+    trv_on_off = config.get(DOMAIN).get(CONF_TRV_ON_OFF)
```

Comment:
Don't write it like this. It will work since we have voluptuous validation already done, but it's bad practice.

Either you don't know if the key is within the dictionary of interest, and you are worried about a `KeyError` if you lookup the key directly. Then you use the `get` method of the dictionary. You can put a default value to return if the key is missing.
```py
value = config.get("key", "default_value")
```
Or you are sure that the key exists in the dictionary, or you want to get a possible `KeyError`, and you just lookup directly.
```py
value = config["key"]
```
In this case where you need to dig into a nested dictionary, you have two options:
If you want to avoid errors and are not sure before about the contents of the dicts, the default value of an empty dict in the first lookup makes sure we don't error when doing the next lookup. The *default* default value if not specified, is `None`. That would give an error if trying to use `get` on `None`.
```py
trv_on_off = config.get(DOMAIN, {}).get(CONF_TRV_ON_OFF, False)
```
or, when you are sure about the contents, like in our case where we have already done voluptuous validation of the config:
```py
trv_on_off = config[DOMAIN][CONF_TRV_ON_OFF]
```